tests/test_cli.py

```diff
@@ -444,3 +444,55 @@ def test_cli_database_error(self, strict_writer, all_db_checkpoint_manager, capf
 
         expected_re = re.compile('Stopping because of database error')
         assert re.search(expected_re, out)
+
+
+# An input where missing fields should be added due to declared data types.
+DATA_TYPES_CLIENT = MockCommCareHqClient({
+    'form': [
+        (
+            {'limit': DEFAULT_BATCH_SIZE, 'order_by': ['server_modified_on', 'received_on']},
+            [
+                {'id': 1, 'form': {}},
+                {'id': 2, 'form': {}}
+            ]
+        ),
+    ],
+})
+
+@pytest.mark.dbtest
+class TestCLIWithDataTypes(object):
+    def test_cli_data_types_add_columns(self, strict_writer, all_db_checkpoint_manager, capfd):
+        args = make_args(
+            query='tests/014_ExportWithDataTypes.xlsx',
+            output_format='sql'
+        )
+        # set this so that it get's written to the checkpoints
+        checkpoint_manager.query = args.query
+
+        api_client_patch = mock.patch('commcare_export.cli._get_api_client',
+                                      return_value=DATA_TYPES_CLIENT)
+        # have to mock these to override the pool class otherwise they hold the db connection open
+        strict_writer_patch = mock.patch('commcare_export.cli._get_writer',
+                                         return_value=strict_writer)
+        checkpoint_patch = mock.patch('commcare_export.cli._get_checkpoint_manager',
+                                      return_value=all_db_checkpoint_manager)
```

Comment:
did you have to figure all these mocks out or were they duplicated from another test (if the latter it might be good to try and unify them)?

Reply:
Good point. Since a lot of the code was like the _pull_data function, I factored it out into a new _pull_mock_data function.